pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -134,13 +137,28 @@ func (pl *CSILimits) Filter(ctx context.Context, _ *framework.CycleState, pod *v
 }
 
 func (pl *CSILimits) filterAttachableVolumes(
-	csiNode *storagev1.CSINode, volumes []v1.Volume, namespace string, result map[string]string) error {
+	pod *v1.Pod, csiNode *storagev1.CSINode, volumes []v1.Volume, namespace string, result map[string]string) error {
 	for _, vol := range volumes {
-		// CSI volumes can only be used as persistent volumes
-		if vol.PersistentVolumeClaim == nil {
+		// CSI volumes can only be used through a PVC.
+		pvcName := ""
+		ephemeral := false
+		switch {
+		case vol.PersistentVolumeClaim != nil:
+			pvcName = vol.PersistentVolumeClaim.ClaimName
+		case vol.Ephemeral != nil:
+			if !utilfeature.DefaultFeatureGate.Enabled(features.GenericEphemeralVolume) {
```

Comment:
we recently started trying to remove dependencies on calling `utilfeature.DefaultFeatureGate.Enabled()` directly in plugins, instead going with a pattern that shims in the featuregate flag setting as a parameter in the plugin registry. See https://github.com/kubernetes/kubernetes/pull/98583 and https://github.com/kubernetes/kubernetes/pull/98446 for examples

Reply:
I'll change to that.